cmd/update/volume_move.go

```diff
@@ -0,0 +1,378 @@
+	return nil
+}
+
+func (c *volumeMoveCommand) waitForReplicaOnNode(ctx context.Context, nsID id.Namespace) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out while watching for desired state but the operation may still be running (server side)")
+
+		case <-time.After(time.Second * 5):
+			vol, err := c.getVolume(ctx, nsID)
+			if err != nil {
+				return fmt.Errorf("failed to get updated volume config from api: %s", err)
+			}
+
+			// find the new deployment added with this operation
+			for _, rep := range vol.Replicas {
+				if rep.Node.String() == c.toNode {
+					c.display.TextMessage(ctx, c.writer, fmt.Sprintf("new replica added on node %s, starting sync", c.toNode))
+					return nil
+				}
+			}
+		}
+	}
+}
+
+func (c *volumeMoveCommand) waitForReplicaOnNodeToSync(ctx context.Context, nsID id.Namespace) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out while watching for desired state but the operation may still be running (server side)")
+
+		case <-time.After(time.Second * 5):
+			vol, err := c.getVolume(ctx, nsID)
+			if err != nil {
+				return fmt.Errorf("failed to get updated volume config from api: %s", err)
+			}
+
+			// find the new deployment added with this operation
+			for _, rep := range vol.Replicas {
+				if rep.Node.String() == c.toNode {
+					if rep.Health == health.ReplicaReady {
+						return nil
+					} else {
+						c.display.ReplicaSyncProgress(ctx, c.writer, *rep.SyncProgress)
+					}
+				}
+			}
+		}
+	}
+}
+
+func (c *volumeMoveCommand) waitForMasterPromote(ctx context.Context, nsID id.Namespace) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out while watching for desired state but the operation may still be running (server side)")
+
+		case <-time.After(time.Second * 5):
+			vol, err := c.getVolume(ctx, nsID)
+			if err != nil {
+				return fmt.Errorf("failed to get updated volume config from api: %s", err)
+			}
+
+			if vol.Master.Node.String() == c.toNode && vol.Master.Health == health.MasterOnline {
+				return nil
+			}
+		}
+	}
+}
+
+func (c *volumeMoveCommand) getVolume(ctx context.Context, nsID id.Namespace) (vol *model.Volume, err error) {
+	useIDs, err := c.config.UseIDs()
+	if err != nil {
+		return nil, err
+	}
+
+	if useIDs {
+		vol, err = c.client.GetVolume(ctx, nsID, id.Volume(c.volumeID))
+		if err != nil {
+			return nil, err
+		}
+		return vol, nil
+	} else {
+		vol, err = c.client.GetVolumeByName(ctx, nsID, c.volumeID)
+		if err != nil {
+			return nil, err
+		}
+		return vol, nil
+	}
+}
+
+func newMoveCmd(w io.Writer, client interfaces.Client, config interfaces.ConfigProvider) *cobra.Command {
+	c := &volumeMoveCommand{
+		config: config,
+		client: client,
+		writer: w,
+	}
+
+	cobraCommand := &cobra.Command{
+		Use:   "move [volume name] [source node] [destination node]",
+		Short: "Updates a volume's placement",
+		Example: `
```

Comment:
I think the example is no longer correct. 

It should be this right?
```
storageos update volume move my-volume-name --source-node-id oldNodeID --destination-node-id newNodeID --namespace my-namespace-name
``` 

i.e using flags